realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -130,8 +132,19 @@ class JniConfigWrapper {
             user = SyncManager::shared().get_user(user_identity, refresh_token,
                                                   realm::util::Optional<std::string>(realm_auth_url));
         }
-        m_config.sync_config = std::make_shared<SyncConfig>(SyncConfig{
-            user, realm_url, SyncSessionStopPolicy::Immediately, std::move(bind_handler), std::move(error_handler)});
+        if (sync_ssl_trust_certificate_path) {
```

Comment:
I would probably go with the `util:none` approach in this case, especially since the list of parameters are so long